app/src/androidTest/java/com/github/houseorganizer/houseorganizer/FirebaseTestsHelper.java

```diff
@@ -0,0 +1,89 @@
+package com.github.houseorganizer.houseorganizer;
+
+import com.google.android.gms.tasks.Task;
+import com.google.android.gms.tasks.Tasks;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.QuerySnapshot;
+import com.google.firebase.storage.FirebaseStorage;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+/**
+ *
+ * All functions in this class are SYNCHRONOUS
+ *
+ */
+public class FirebaseTestsHelper {
+
+    private static boolean authEmulatorActivated = false;
+    private static boolean firebaseEmulatorActivated = false;
+    private static boolean databaseEmulatorActivated = false;
+
+    public static String TEST_USER_MAIL = "test@house.com";
+    public static String TEST_USER_PASSWORD = "Householder";
+    public static String TEST_HOUSEHOLD_NAME = "MyFavoriteHouse";
+
+    public static void startAuthEmulator(){
+        if(authEmulatorActivated) return;
+        FirebaseAuth.getInstance().useEmulator("10.0.2.2", 9099);
+    }
+
+    public static void startFirestoreEmulator(){
+        if(firebaseEmulatorActivated) return;
+        FirebaseFirestore.getInstance().useEmulator("10.0.2.2", 8080);
+        firebaseEmulatorActivated = true;
+    }
+
+    public static void startStorageEmulator(){
+        if(databaseEmulatorActivated) return;
+        FirebaseStorage.getInstance().useEmulator("10.0.2.2", 9199);
+        databaseEmulatorActivated = true;
+    }
+
+    public static void createFirebaseTestUser() throws ExecutionException, InterruptedException {
+
+        Task<AuthResult> t = FirebaseAuth.getInstance().signInWithEmailAndPassword(TEST_USER_MAIL, TEST_USER_PASSWORD)
+                .addOnCompleteListener(task -> {
+                    if(task.getResult().getUser()==null){
+                        FirebaseAuth.getInstance().createUserWithEmailAndPassword(TEST_USER_MAIL, TEST_USER_PASSWORD);
+                    }else{
+                        FirebaseAuth.getInstance().signOut();
+                    }
+                });
+        Tasks.await(t);
```

Comment:
Are we sure the await terminates or time outs ?

Reply:
If it doesn't terminates the test will be written as failed at some point so I think this is not an issue. On the other hand if it is prefered I can use the awaits(task, timeout) surcharge so we are sure it does terminates.

Reply:
I think it would be better actually yes